updatersync/src/main/java/nl/openweb/hippo/groovy/watch/FileSystemWatcher.java

```diff
@@ -290,37 +298,45 @@ void start() {
+            }
+        }
+
+        private void processDirectoryChange(final WatchEvent.Kind<?> kind, final Path changedAbsPath) {
+            if (kind == StandardWatchEventKinds.ENTRY_CREATE) {
+                listener.directoryCreated(changedAbsPath);
+            } else if (kind == StandardWatchEventKinds.ENTRY_MODIFY) {
+                listener.directoryModified(changedAbsPath);
+            } else if (kind == StandardWatchEventKinds.ENTRY_DELETE) {
+                listener.directoryDeleted(changedAbsPath);
+            } else if (kind == StandardWatchEventKinds.OVERFLOW) {
+                if (changedAbsPath.toFile().exists()) {
+                    log.info("Having an event overflow for '{}'. Entire directory '{}' will be recreated",
+                            changedAbsPath, changedAbsPath);
                     listener.directoryCreated(changedAbsPath);
-                } else if (kind == StandardWatchEventKinds.ENTRY_MODIFY) {
-                    listener.directoryModified(changedAbsPath);
-                } else if (kind == StandardWatchEventKinds.ENTRY_DELETE) {
+                } else {
+                    log.info("Having an event overflow for non existing directory '{}'. Directory '{}' will be removed",
+                            changedAbsPath, changedAbsPath);
                     listener.directoryDeleted(changedAbsPath);
-                } else if (kind == StandardWatchEventKinds.OVERFLOW) {
-                    if (Files.exists(changedAbsPath)) {
-                        log.info("Having an event overflow for '{}'. Entire directory '{}' will be recreated",
-                                changedAbsPath, changedAbsPath);
-                        listener.directoryCreated(changedAbsPath);
-                    } else {
-                        log.info("Having an event overflow for non existing directory '{}'. Directory will be removed",
-                                changedAbsPath, changedAbsPath);
-                        listener.directoryDeleted(changedAbsPath);
-                    }
-                }
-            } else {
-                if (kind == StandardWatchEventKinds.ENTRY_CREATE) {
-                    listener.fileCreated(changedAbsPath);
-                } else if (kind == StandardWatchEventKinds.ENTRY_MODIFY) {
-                    listener.fileModified(changedAbsPath);
-                } else if (kind == StandardWatchEventKinds.ENTRY_DELETE) {
-                    listener.fileDeleted(changedAbsPath);
-                } else if (kind == StandardWatchEventKinds.OVERFLOW) {
-                    throw new IllegalStateException("Only a directory should even possibly overflow in events, for example" +
-                            " by saving 1000 new files in one go.");
                 }
             }
         }
 
+        private void processFileChange(final WatchEvent.Kind<?> kind, final Path changedAbsPath) {
+            if (kind == StandardWatchEventKinds.ENTRY_CREATE) {
```

Comment:
Could be rewritten to switch statement

Reply:
Not really a good idea, A switch works with the byte, short, char, and int primitive data types, some wrappers, and Enum entries. This doesn't apply to this check.